frodo.js

```diff
@@ -1,98 +1,108 @@
 /**
```

Comment:
SCRIPT-This new feature lacks unit tests. Could you add some to ensure everything works correctly?